whisper/src/lib.rs

```diff
@@ -706,6 +708,25 @@ fn file_merge<F1: Read + Seek, F2: Read + Write + Seek>(fh_src: &mut F1, fh_dst:
     Ok(())
 }
 
+pub fn merge_all(path_src: &Path, path_dst: &Path, now: u32) -> Result<(), io::Error> {
+    let mut src = WhisperFile::open(path_src)?;
+    let mut dst = WhisperFile::open(path_dst)?;
+
+    if src.info().archives != dst.info().archives {
+        return Err(io::Error::new(io::ErrorKind::Other, "Archive configurations are unalike. Resize the input before merging"));
+    }
+
+    let archives = src.info().archives
+        .clone().into_iter()
+        .sorted_by_key(|archive| archive.retention());
+
+    for archive in &archives {
+        let points = src.dump(archive.seconds_per_point)?;
```

Comment:
Dump should be filtered here: null values are converted to Point{0, 0}, so they should be filter out.

In whisper-resize I added derive Default to Point and then filter out it.

https://github.com/GiantPlantsSociety/graphite-rs/pull/45/files#diff-dcaf987c869b88d2c9de4e8c36cd3cc9R153

Reply:
Also there is an idea, the next level archive interval should not include previous archive interval. Otherwise aggregated value can overwrite more precise value.